docs/appendices/release-notes/unreleased.rst

```diff
@@ -225,3 +225,6 @@ Fixes
 - Fixed casts of ``TEXT`` to ``REGCLASS`` data types which were resulting in
   wrong ``REGCLASS`` values as the  ``current_schema`` was not taken into
   account.
+
+- Fixed an issue which caused ``PRIMARY KEY`` columns to be shown
```

Comment:
Not sure we need a change entry, let alone backporting it - as discussed, actual motivation is to align internal state and pg_catalog thingy is more like side effect

Reply:
It doesn't hurt to add an entry here, and I'd also vote to backport it.

Reply:
Also think so taking to account now we have simpler approach (no explicit not-nulls in metadata, fix only `isNullable() `property).